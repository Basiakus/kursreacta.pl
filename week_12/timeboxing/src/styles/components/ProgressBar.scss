@import '../variables.scss';
@import '../utilities.scss';


.ProgressBar {
   box-sizing: border-box;
   border: 1px solid black;
   border-radius: $default-border-radius;
   margin-bottom: 20px;
   height: 25px;
   padding: 4px;
   display: flex;
   align-items: center;
   width: 100%;
   &--big {
      height: 50px;
   }
   &--borderBlue {
      border: 2px solid lighten(rgb(46, 102, 213), 10);
   }
}
