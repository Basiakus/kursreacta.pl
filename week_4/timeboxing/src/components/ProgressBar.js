import React from 'react';
import '../styles/components/ProgressBar.scss';

function ProgressBar({ percent = 0, trackRemaining, className }) {
   return (
      <div
         className={`ProgressBar ${className}`}
         style={{
            background: (trackRemaining ?
               `linear-gradient(orangered, orangered) right /${percent}% no-repeat content-box` :
               `linear-gradient(orangered, orangered) left /${percent}% no-repeat content-box`
            )
         }}
      >
      </div>
   )
}

export default ProgressBar;