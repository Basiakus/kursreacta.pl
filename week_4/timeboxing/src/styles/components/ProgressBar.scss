@import '../variables.scss';
@import '../utilities.scss';


.ProgressBar {
   box-sizing: border-box;
   border: 1px solid orangered;
   border-radius: $default-border-radius;
   margin-bottom: 20px;
   height: 25px;
   padding: 4px;
   display: flex;
   align-items: center;
   width: 100%;
}