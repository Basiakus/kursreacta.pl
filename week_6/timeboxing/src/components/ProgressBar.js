import React from 'react';
import classNames from 'classnames';
//import '../styles/components/ProgressBar.scss';

function ProgressBar({ percent = 0, trackRemaining, barColor, borderBlue, isBig, className}) {
   //`ProgressBar ProgressBar--big ProgressBar--borderBlue`
   let progressBarClassNames = classNames(
      "ProgressBar",
      className,
      {
         "ProgressBar--big": isBig,
         "ProgressBar--borderBlue": borderBlue
      }
   );
   return (
      <div
         className={progressBarClassNames}
         style={{
            background: (trackRemaining ?
               `linear-gradient(${barColor}, ${barColor}) right /${percent}% no-repeat content-box` :
               `linear-gradient(${barColor}, ${barColor}) left /${percent}% no-repeat content-box`
            )
         }}
      >
      </div>
   )
}

export default ProgressBar;