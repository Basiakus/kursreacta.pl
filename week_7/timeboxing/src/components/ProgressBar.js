import React from 'react';
import classNames from 'classnames';
//import '../styles/components/ProgressBar.scss';


const correctSrtingColorValue = function (props, propName, componentName) {
   const propToCheck = props[propName];
   if (propToCheck !== 'red' && propToCheck !== 'blue' && propToCheck !== 'green') {
      return new Error(`niewłaściwy prop ${propName} w komponencie ${componentName}. Opcjonalne wartości to: "green", "blue", "red"`);
   }
}
const correctPercentValue = function (props, propName, componentName) {
   const propToCheck = props[propName];
   if (propToCheck < 0 || propToCheck > 100 || propToCheck === undefined || propToCheck === null) {
      return new Error(`niewłaściwy prop ${propName} w komponencie ${componentName}. Powinien on akceptować tylko liczby, których wartości są pomiędzy 0 a 100. Nie powinien akceptować ani null ani undefined.`);
   }
}

ProgressBar.propTypes = {
   barColor: correctSrtingColorValue,
   percent: correctPercentValue
}

function ProgressBar({ percent, trackRemaining, barColor, borderBlue, isBig, className}) {
   //`ProgressBar ProgressBar--big ProgressBar--borderBlue`
   let progressBarClassNames = classNames(
      "ProgressBar",
      className,
      {
         "ProgressBar--big": isBig,
         "ProgressBar--borderBlue": borderBlue
      }
   );
   return (
      <div
         className={progressBarClassNames}
         style={{
            background: (trackRemaining ?
               `linear-gradient(${barColor}, ${barColor}) right /${percent}% no-repeat content-box` :
               `linear-gradient(${barColor}, ${barColor}) left /${percent}% no-repeat content-box`
            )
         }}
      >
      </div>
   )
}

export default ProgressBar;