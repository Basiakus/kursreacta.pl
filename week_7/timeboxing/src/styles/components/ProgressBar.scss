@import '../variables.scss';
@import '../utilities.scss';


.ProgressBar {
   box-sizing: border-box;
   border: 1px solid black;
   border-radius: $default-border-radius;
   margin-bottom: 20px;
   height: 25px;
   padding: 4px;
   display: flex;
   align-items: center;
   width: 100%;
}
.ProgressBar--big {
   height: 50px;
}
.ProgressBar--borderBlue {
   border: 2px solid lighten(rgb(46, 102, 213), 10);
}