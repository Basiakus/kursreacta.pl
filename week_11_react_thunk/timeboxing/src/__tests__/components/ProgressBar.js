import React from 'react';
import ProgressBar from '../../components/ProgressBar.js';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
var root = null;
let progressBarRenerer = null;
describe('TEST <ProgressBar /> Component', () => {
     beforeEach(() => {
          root = document.createElement('div');
          ReactDOM.render(
               <ProgressBar barColor="blue" percent={20} trackRemaining={false}/>, root
          )
     })
     xit('test for corrent className', () => {
          expect(root.childNodes[0].className).toEqual('ProgressBar');
     });
});

describe('Use renderer on <ProgressBar />', () => {
     beforeEach(() => {
          progressBarRenerer = renderer.create(
               <ProgressBar barColor="blue" percent={11} trackRemaining={false} />
          )
     })
     it('Create snapshot file represent a component JSON file', () => {
          console.log(progressBarRenerer.toJSON())
          expect(progressBarRenerer.toJSON()).toMatchSnapshot();
     })
     it('check value of percent if is between 0-100', () => {
          expect(progressBarRenerer.toJSON().props.style.background).toMatch(/[0-100]/)
     })
});