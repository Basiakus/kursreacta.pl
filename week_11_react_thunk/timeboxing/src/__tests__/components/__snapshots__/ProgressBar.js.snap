// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use renderer on <ProgressBar /> Create snapshot file represent a component JSON file 1`] = `
<div
  className="ProgressBar"
  style={
    Object {
      "background": "linear-gradient(blue, blue) left /11% no-repeat content-box",
    }
  }
/>
`;
